### PR TITLE
Use new Blaze API.

### PR DESCRIPTION
**Pull request by james-s-willis**

### CODE REVIEW

File: tools/docker/Dockerfile [hunk 1]
```diff
@@ -56,7 +56,7 @@ RUN git clone https://github.com/kiyo-masui/bitshuffle.git bitshuffle && \
 RUN apt-get update && \
     apt-get -y install libopenblas-dev liblapack-dev liblapacke-dev && \
     apt-get clean && apt-get autoclean && \
-    git clone https://bitbucket.org/blaze-lib/blaze.git blaze
+    git clone https://bitbucket.org/blaze-lib/blaze.git blaze && cd blaze && git pull && cd ..
```
Comment (andrerenard) [inline tools/docker/Dockerfile:59]: We should add this to the MacOS build as well just in case it's also caching things.

**Comment by andrerenard** [comment]
Actually while we are at it, would you mind checking other places where we clone without doing a pull?  In theory this could become a problem again for another include.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -236,21 +236,24 @@ jobs:
...
+        git clone https://github.com/h5py/h5py.git h5py && cd h5py && git pull
         python3 setup.py configure --hdf5=/usr/local/opt/hdf5@1.10/
         python3 setup.py build
 
     - name: Install bitshuffle
       env:
         HDF5_DIR: /usr/local/opt/hdf5@1.10/
       run: |
-        git clone https://github.com/kiyo-masui/bitshuffle.git bitshuffle && cd bitshuffle
+        git clone https://github.com/kiyo-masui/bitshuffle.git bitshuffle && \
```
Comment (andrerenard) [inline .github/workflows/main.yml:247]:
```suggestion
        git clone https://github.com/kiyo-masui/bitshuffle.git bitshuffle
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -236,21 +236,24 @@ jobs:
...
+        cd bitshuffle && git pull
         python3 setup.py install --h5plugin --h5plugin-dir=/usr/local/opt/hdf5@1.10/lib/plugin
 
     - name: Clone Blaze and HighFive
       run: |
-        git clone https://bitbucket.org/blaze-lib/blaze.git blaze
-        git clone --single-branch --branch extensible-datasets https://github.com/jrs65/HighFive.git
+        git clone https://bitbucket.org/blaze-lib/blaze.git blaze && \
```
Comment (andrerenard) [inline .github/workflows/main.yml:253]:
```suggestion
        git clone https://bitbucket.org/blaze-lib/blaze.git blaze
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -236,21 +236,24 @@ jobs:
...
+        cd bitshuffle && git pull
         python3 setup.py install --h5plugin --h5plugin-dir=/usr/local/opt/hdf5@1.10/lib/plugin
 
     - name: Clone Blaze and HighFive
       run: |
-        git clone https://bitbucket.org/blaze-lib/blaze.git blaze
-        git clone --single-branch --branch extensible-datasets https://github.com/jrs65/HighFive.git
+        git clone https://bitbucket.org/blaze-lib/blaze.git blaze && \ 
+        cd blaze && git pull && cd ..
+        git clone --single-branch --branch extensible-datasets https://github.com/jrs65/HighFive.git && \
```
Comment (andrerenard) [inline .github/workflows/main.yml:255]:
```suggestion
        git clone --single-branch --branch extensible-datasets https://github.com/jrs65/HighFive.git
```

**Comment by andrerenard** [comment]
For future reference the blaze change is: https://bitbucket.org/blaze-lib/blaze/commits/8e32af9ad79405048b8918786e628c7ca29d4f3f#chg-blaze/math/dense/DenseVector.h

**Comment by jrs65** [comment]
This is fine, but I think I'd preferred to have just pinned the version of blaze installed by the CI server to a working release for the short term. We probably should have been doing that anyway. In future please tag me for review for changes to the eigenvis code.

**Pull request merged**